File: 082/cpp/main.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode *root = new ListNode();
        ListNode *temp1 = root;
        ListNode *temp2 = head;
        bool dup = false;
        for (; temp2 != nullptr && temp2->next != nullptr; temp2 = temp2->next) {
            if (temp2->val == temp2->next->val) {
                dup = true;
            } else {
                if (!dup) {
                    temp1->next = new ListNode(temp2->val);
                    temp1 = temp1->next;
                }
                dup = false;
            }
        }
        if (temp2 != nullptr && !dup) {
            temp1->next = new ListNode(temp2->val);
        }
        return root->next;
    }
};

File: 082/rust/main.rs
// Definition for singly-linked list.
// #[derive(PartialEq, Eq, Clone, Debug)]
// pub struct ListNode {
//   pub val: i32,
//   pub next: Option<Box<ListNode>>
// }
// 
// impl ListNode {
//   #[inline]
//   fn new(val: i32) -> Self {
//     ListNode {
//       next: None,
//       val
//     }
//   }
// }
impl Solution {
    pub fn delete_duplicates(head: Option<Box<ListNode>>) -> Option<Box<ListNode>> {
        let mut head = head;
        let mut root = ListNode::new(0);
        let mut temp = &mut root;
        let mut val = -200;
        let mut cnt = false;
        while head.is_some() {
            let t = head.unwrap();
            if t.val == val {
                cnt = true;
            } else {
                if val != -200 && !cnt {
                    temp.next = Some(Box::new(ListNode::new(val)));
                    temp = temp.next.as_mut().unwrap();
                }
                val = t.val;
                cnt = false;
            }
            head = t.next;
        }
        if val != -200 && !cnt {
            temp.next = Some(Box::new(ListNode::new(val)));
        }
        return root.next;
    }
}
